### PR TITLE
[JENKINS-36720] - Fix Spotbugs issues (SE_COMPARATOR_SHOULD_BE_SERIALIZABLE, ES_COMPARING_STRINGS_WITH_EQ)

### PR DESCRIPTION
**Pull request by StefanSpieker**
Fixed some Spotbugs issues 
One of Type ES_COMPARING_STRINGS_WITH_EQ
And some of [SE_COMPARATOR_SHOULD_BE_SERIALIZABLE](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#se-comparator-doesn-t-implement-serializable-se-comparator-should-be-serializable).
I also added the SerialVersionUID to the comparators, that would otherwise trigger the next Spotbugs error.

If we get this in, the number of spotbugs warnings will be less then 400 😁

See [JENKINS-36720](https://issues.jenkins-ci.org/browse/JENKINS-36720).

### Proposed changelog entries

* N/A

### CODE REVIEW

File: core/src/main/java/hudson/Functions.java [hunk 1]
```diff
@@ -25,6 +25,7 @@
  */
 package hudson;
 
+import com.google.inject.internal.asm.$ByteVector;
```
Comment (jvz) [inline core/src/main/java/hudson/Functions.java]: This doesn't look like it belongs here.

Comment (StefanSpieker) [reply core/src/main/java/hudson/Functions.java]: Thanks. I don't know how it got there. I removed it. Thanks for pointing that out.

File: core/src/main/java/hudson/Functions.java [hunk 2]
```diff
@@ -1306,7 +1309,7 @@ public static ThreadGroupMap sortThreadsAndGetGroupMap(ThreadInfo[] list) {
     private static class ThreadSorterBase {
         protected Map<Long,String> map = new HashMap<>();
 
-        private ThreadSorterBase() {
+        public ThreadSorterBase() {
```
Comment (MRamonLeon) [inline core/src/main/java/hudson/Functions.java]: out of curiosity: why did you change it?

**Review by MRamonLeon — APPROVED** [review]

File: core/src/main/java/hudson/Functions.java [hunk 3]
```diff
@@ -1326,7 +1329,9 @@ protected int compare(long idA, long idB) {
         }
     }
 
-    public static class ThreadGroupMap extends ThreadSorterBase implements Comparator<ThreadInfo> {
+    public static class ThreadGroupMap extends ThreadSorterBase implements Comparator<ThreadInfo>, Serializable {
+
+        private static final long serialVersionUID = 7803975728695308444L;
```
Comment (rsandell) [inline core/src/main/java/hudson/Functions.java]: For some reason, the standard in this code base is to default the `serialVersionUID` to `1L` instead of some randomly generated number. The reason for it is unknown to me, but I've gotten scolded for this by others in the past :)

**Review by rsandell — APPROVED** [review]

File: core/src/main/java/hudson/model/TimeZoneProperty.java [hunk 4]
```diff
@@ -105,7 +105,7 @@ public static String forCurrentUser() {
         }
 
         TimeZone tz = TimeZone.getTimeZone(tzp.timeZoneName);
-        if (tz.getID() != tzp.timeZoneName) {
+        if (!tz.getID().equals(tzp.timeZoneName)) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/TimeZoneProperty.java]: That one might be a real defect in some JVM implementations. Should we changelog it?

**Review by oleg-nenashev — APPROVED** [review]
I plan to merge it today once we cleanup the changelog side

**Comment by oleg-nenashev** [comment]
FYI @StefanSpieker I am going to proceed without a changelog for now. I do not see relevant defects in JIRA, and I doubt it would appear in real life. If you would like to add an entry, please let me know

File: core/src/main/java/hudson/Functions.java (hunk 2 above)
Comment (StefanSpieker) [reply core/src/main/java/hudson/Functions.java]: it needed a public constructor to be serializable.
